resources/services/ssm/associations.go

```diff
@@ -0,0 +1,319 @@
+				Type:        schema.TypeJSON,
+				Resolver:    resolveAssociationTargets,
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:          "aws_ssm_association_target_locations",
+				Description:   "The combination of Amazon Web Services Regions and Amazon Web Services accounts targeted by the current Automation execution.",
+				Resolver:      fetchSsmAssociationTargetLocations,
+				IgnoreInTests: true,
+				Columns: []schema.Column{
+					{
+						Name:        "association_cq_id",
+						Description: "Unique CloudQuery ID of aws_ssm_associations table (FK)",
+						Type:        schema.TypeUUID,
+						Resolver:    schema.ParentIdResolver,
+					},
+					{
+						Name:        "accounts",
+						Description: "The Amazon Web Services accounts targeted by the current Automation execution.",
+						Type:        schema.TypeStringArray,
+					},
+					{
+						Name:        "execution_role_name",
+						Description: "The Automation execution role used by the currently running Automation",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "regions",
+						Description: "The Amazon Web Services Regions targeted by the current Automation execution.",
+						Type:        schema.TypeStringArray,
+					},
+					{
+						Name:        "target_location_max_concurrency",
+						Description: "The maximum number of Amazon Web Services Regions and Amazon Web Services accounts allowed to run the Automation concurrently.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "target_location_max_errors",
+						Description: "The maximum number of errors allowed before the system stops queueing additional Automation executions for the currently running Automation.",
+						Type:        schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchSsmAssociations(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().SSM
+	params := ssm.ListAssociationsInput{}
+	for {
+		result, err := svc.ListAssociations(ctx, &params, func(o *ssm.Options) { o.Region = cl.Region })
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		for _, a := range result.Associations {
+			desc, err := svc.DescribeAssociation(
+				ctx,
+				&ssm.DescribeAssociationInput{
+					AssociationId:      a.AssociationId,
+					AssociationVersion: a.AssociationVersion,
+				},
+				func(o *ssm.Options) { o.Region = cl.Region },
+			)
+			if err != nil {
+				return diag.WrapError(err)
+			}
+			if desc.AssociationDescription != nil {
+				res <- *desc.AssociationDescription
+			}
+		}
```

Comment:
Not sure how many Associations accounts typically have. Do we want to put the DescribeAssociation call in a go routine? We have done it for a couple of instances like here:
https://github.com/cloudquery/cq-provider-aws/pull/701/files

Reply:
Made the calls concurrent as suggested.